crates/worker/src/actor/spawner.rs

```diff
@@ -176,8 +188,61 @@ where
         W::Input: Serialize + for<'de> Deserialize<'de>,
         W::Output: Serialize + for<'de> Deserialize<'de>,
     {
-        let worker = DedicatedWorker::new(loader_path).expect("failed to spawn worker");
+        self.spawn_with_options(SpawnOptions::new(loader_path).with_loader())
+            .expect("failed to spawn worker")
+    }
+
+    /// Spawns a Worker according to the given options. Returns `None` if failed.
+    pub fn spawn_with_options(&self, options: SpawnOptions<'_>) -> Option<WorkerBridge<W>>
+    where
+        W::Input: Serialize + for<'de> Deserialize<'de>,
+        W::Output: Serialize + for<'de> Deserialize<'de>,
+    {
+        let worker = create_worker(options)?;
 
-        self.spawn_inner(worker)
+        Some(self.spawn_inner(worker))
+    }
+}
+
+/// Options for spawning a worker.
+#[derive(Debug, Copy, Clone)]
+pub struct SpawnOptions<'a> {
+    path: &'a str,
+    with_loader: bool,
+    as_module: bool,
+}
+impl<'a> SpawnOptions<'a> {
+    /// Creates a default [SpawnOptions].
+    pub fn new(path: &'a str) -> Self {
+        Self {
+            path,
+            with_loader: false,
+            as_module: false,
+        }
+    }
+
+    /// Indicates that the `path` is a loader shim script (e.g. when using trunk,
+    /// created by using the
+    /// [`data-loader-shim`](https://trunkrs.dev/assets/#link-asset-types)
+    /// asset type) and one does not need to be generated when creating the
+    /// worker.
+    pub fn with_loader(mut self) -> Self {
+        self.with_loader = true;
+        self
+    }
+
+    /// The worker will be spawned with
+    /// [`options.type`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker#type)
+    /// set to `module`. This is currently supported on all browsers except
+    /// Safari (see compatibility table at above link). If it is not supported,
```

Comment:
I have tested this and seems like Safari does support `{ type: ‘module' }`.

The MDN documentation might be outdated as it indicates Safari as supporting ESModule Worker but not `{type: ‘module’}`. However, it is not possible to create ESModule Worker without supporting `type` first.

Reply:
Shouldn't have blindly trusted the MDN doc, will test myself as well and update